packages/url_launcher/url_launcher_web/lib/url_launcher_web.dart

```diff
@@ -43,6 +47,9 @@ class UrlLauncherPlugin extends UrlLauncherPlatform {
   /// Registers this class as the default instance of [UrlLauncherPlatform].
   static void registerWith(Registrar registrar) {
     UrlLauncherPlatform.instance = UrlLauncherPlugin();
+    linkDelegate = (LinkInfo linkInfo) => WebLinkDelegate(linkInfo);
+    // ignore: undefined_prefixed_name
+    ui.platformViewRegistry.registerViewFactory(viewType, linkViewFactory);
```

Comment:
_In TV Salesman Voice:_ You could hide all this logic, if the LinkDelegate was a PlatformInterface! :P And just leave it as: `LinkPlatform.instance = WebLinkPlatform();` :P

Reply:
(I understand this is not going to stay as a plugin forever, so... whatever makes sense for the framework/engine is probably best :P)

Reply:
I would love to make it work like you described but I couldn't. The difference in my case is that the default implementation of "LinkPlatform" has to live in `url_launcher`, not in `url_launcher_platform_interface`. And AFAICT, there's no way to set `LinkPlatform.instance = DefaultLinkPlatform();` in `url_launcher`. I hope I'm wrong here, and hope there's a way to do it :)